R/pkg/inst/worker/worker.R

```diff
@@ -100,7 +109,24 @@ if (isEmpty != 0) {
     # Timing reading input data for execution
     inputElap <- elapsedSecs()
 
-    output <- computeFunc(partition, data)
+    if (isDataFrame) {
+      if (deserializer == "row") {
+        # Transform the list of rows into a data.frame
+        data <- do.call(rbind.data.frame, c(data, stringsAsFactors = FALSE))
+        names(data) <- colNames
+      } else {
+        # Check to see if data is a valid data.frame
+        stopifnot(class(data) == "data.frame")
+      }
+      output <- computeFunc(data)
+      if (serializer == "row") {
+        # Transform the result data.frame back to a list of rows
+        output <- split(output, seq(nrow(output)))
+      }
```

Comment:
Can we assert that the serializer is `byte` in the else clause ? Thats what we expect it to be right ?